colcon_core/environment/path.py

```diff
@@ -45,3 +46,41 @@ def _create_environment_hooks(
                 break
 
         return hooks
+
+
+class PythonScriptsPathEnvironment(EnvironmentExtensionPoint):
+    """Extend the `PATH` variable to find python scripts."""
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(
+            EnvironmentExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+
+    def create_environment_hooks(self, prefix_path, pkg_name):  # noqa: D102
+        hooks = self._create_environment_hooks(prefix_path, pkg_name, 'bin')
+        if sys.platform == 'win32':
+            hooks += self._create_environment_hooks(
+                prefix_path, pkg_name, 'Scripts', '-scripts')
```

Comment:
This is copy/paste cruft from the original `PathEnvironment` class, and it can actually be removed from there too because `PythonScriptsPathEnvironment` actually obsoletes it. So both of these extensions can be simplified a fair amount.

I verified that `sysconfig.get_path('scripts', vars={'base': '/foo'})` returns `/foo/Scripts` on Windows 11 with Python 3.9.

Reply:
Simplified logic in the recent commits. Let me know what you think!

Reply:
Loads better. Thank you!